From 962e175c8c6d4ed2230a1dd604c8836dc52bca49 Mon Sep 17 00:00:00 2001
From: Haijun Zhang <Haijun.Zhang@freescale.com>
Date: Mon, 12 May 2014 19:09:14 +0800
Subject: [PATCH 1012/1089] mmc/eSDHC: host need long time to generate command
 complete interrupt

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

According to Spec 2.0, command complete interrupt will be generate within
150 SD-CLK. But this was not long enough for P5040 board. So add quirk
SDHCI_QUIRK2_LONG_TIME_CMD_COMPLETE_IRQ to enlarge this detect time to avoid
hardware timeout error.

Signed-off-by: Haijun Zhang <Haijun.Zhang@freescale.com>
Change-Id: I26863bf687275191ead0e6396c381d9c15fb8e07
Reviewed-on: http://git.am.freescale.net:8181/12215
Reviewed-by: Shengzhou Liu <Shengzhou.Liu@freescale.com>
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Jose Rivera <German.Rivera@freescale.com>
---
 drivers/mmc/host/sdhci-pltfm.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/host/sdhci-pltfm.c b/drivers/mmc/host/sdhci-pltfm.c
index 475ed4c..2a9416b 100644
--- a/drivers/mmc/host/sdhci-pltfm.c
+++ b/drivers/mmc/host/sdhci-pltfm.c
@@ -104,7 +104,8 @@ void sdhci_get_of_property(struct platform_device *pdev)
 			host->quirks2 |= SDHCI_QUIRK2_BROKEN_TRIM;
 		}
 
-		if (of_device_is_compatible(np, "fsl,p5020-esdhc"))
+		if (of_device_is_compatible(np, "fsl,p5020-esdhc") ||
+		    of_device_is_compatible(np, "fsl,p5040-esdhc"))
 			host->quirks2 |= SDHCI_QUIRK2_LONG_TIME_CMD_COMPLETE_IRQ;
 
 		if (of_device_is_compatible(np, "fsl,p5040-esdhc") ||
-- 
2.0.2

